lib/rubocop/cop/rails/reflection_class_name.rb

```diff
@@ -23,14 +25,24 @@ class ReflectionClassName < Cop
         PATTERN
 
         def_node_matcher :reflection_class_name, <<~PATTERN
-          (pair (sym :class_name) [!dstr !str !sym])
+          (pair (sym :class_name) $[!dstr !str !sym])
         PATTERN
 
         def on_send(node)
-          association_with_reflection(node) do |reflection_class_name|
-            add_offense(node, location: reflection_class_name.loc.expression)
+          association_with_reflection(node) do |reflection|
+            add_offense(reflection) do |corrector|
+              class_node = reflection_class_name(reflection)
+              corrector.replace(class_node, replacement(class_node))
+            end
           end
         end
+
+        private
+
+        def replacement(class_node)
+          replacement_node = class_node.send_type? ? class_node.children.first : class_node
```

Comment:
This assumes that the send node is calling `name` on a class, which isn't necessarily the case, e.g. (a bit contrived):

```ruby
%w[accounts admins profiles].each do |things|
  has_many things, class_name: things.classify.prepend("Internal::")
end
```

I think we should only autocorrect a constant or a `name` method call on a constant.

Reply:
@tejasbubane Can you update this one?